Source/src/components/ComponentBillboard.cpp

```diff
@@ -0,0 +1,547 @@
+    {
+        texture = static_cast<ResourceTexture*>(
+            App->resources->GetResource(Resource::Type::TEXTURE, texture_id));
+        has_texture = 1;
+    }
+
+    type = node[BILLBOARD_TYPE].IsDefined() 
+        ? static_cast<BillboardType>(node[BILLBOARD_TYPE].as<int>()) 
+        : BillboardType::HORIZONTAL;
+    
+    play_on_awake = node[BILLBOARD_PLAY_ON_AWAKE].IsDefined() ? 
+        node[BILLBOARD_PLAY_ON_AWAKE].as<bool>() : false;
+
+    x_tiles = 1;
+    if (node[X_TILES].IsDefined() && !node[X_TILES].IsNull())
+    {
+        x_tiles = node[X_TILES].as<int>();
+        x_factor = 1 / (float)x_tiles;
+    }
+
+    y_tiles = 1;
+    if (node[Y_TILES].IsDefined() && !node[Y_TILES].IsNull())
+    {
+        y_tiles = node[Y_TILES].as<int>();
+        y_factor = 1 / (float)y_tiles;
+    }
+
+    billboard_lifetime = node[BILLBOARD_LIFETIME].IsDefined() ?
+        node[BILLBOARD_LIFETIME].as<float>() : 0.0f;
+
+    skip_frames = node[SKIP_FRAMES].IsDefined() ?
+        node[SKIP_FRAMES].as<int>() : 0;
+
+    animation_loop = node[ANIMATION_LOOP].IsDefined() ?
+        node[ANIMATION_LOOP].as<bool>() : false;
+
+    has_flip_x = node[FLIP_X].IsDefined() ?
+        node[FLIP_X].as<bool>() : false;
+
+    has_flip_y = node[FLIP_Y].IsDefined() ?
+        node[FLIP_Y].as<bool>() : false;
+
+    has_color_gradient = node[HAS_COLOR_GRADIENT].IsDefined() ?
+        node[HAS_COLOR_GRADIENT].as<bool>() : false;
+
+    color_cycles = node[COLOR_CYCLES].IsDefined() ? 
+        node[COLOR_CYCLES].as<int>() : 0;
+    
+    color_loop = node[COLOR_LOOP].IsDefined() ?
+        node[COLOR_LOOP].as<bool>() : false;
+
+    if (node[COLOR_GRADIENT].IsDefined())
+    {
+        gradient->clearMarks();
+        for (int i = 0; i < node[COLOR_GRADIENT].size(); ++i)
+        {
+            auto mark = node[COLOR_GRADIENT][i].as<ImGradientMark>();
+            gradient->addMark(mark.position,
+                ImColor(mark.color[0], mark.color[1], mark.color[2], mark.color[3]));
+        }
+    }
+}
+
+void Hachiko::ComponentBillboard::AddTexture()
+{
+    const std::string title = "Select billboard texture ";
+    std::string texture_path;
+    ResourceTexture* res = nullptr;
+
+    if (ImGui::Button("Add Texture"))
+    {
+        ImGuiFileDialog::Instance()->OpenDialog(title.c_str(),
+            "Select Texture", ".png,.tif,.jpg,.tga", "./assets/textures/", 1, nullptr,
+            ImGuiFileDialogFlags_DontShowHiddenFiles | ImGuiFileDialogFlags_DisableCreateDirectoryButton 
+            | ImGuiFileDialogFlags_HideColumnType | ImGuiFileDialogFlags_HideColumnDate);
+    }
+
+    if (ImGuiFileDialog::Instance()->Display(title.c_str()))
+    {
+        if (ImGuiFileDialog::Instance()->IsOk())
+        {
+            texture_path = ImGuiFileDialog::Instance()->GetFilePathName();
+        }
+
+        ImGuiFileDialog::Instance()->Close();
+    }
+
+    texture_path.append(META_EXTENSION);
+    if (!std::filesystem::exists(texture_path.c_str()))
```

Comment:
```suggestion
    if (!std::filesystem::exists(texture_path.c_str()))
```
if (!FileSystem::Exists(texture_path.c_str()))

Reply:
For some reason, FileSystem function is not working, maybe it's only in my pc. I'll investigate more. Thanks